specs/filebeat.spec.yml

```diff
@@ -17,8 +17,10 @@ inputs:
       - redis
     shippers: &shippers
       - shipper
-    command:
-      args: &args
+    command: &command
+      restart_monitoring_period: 5s
+      maximum_restarts_per_period: 1
```

Comment:
I've noticed there is a restart period we can configure that controls how long the process stays in the failed state before it is restarted. We should lower this for Beats that expect to restart on their own.

Defined here, defaults to 10s:
https://github.com/elastic/elastic-agent/blob/668fa22918fa330db979fc9fbd81ef3854deb4c2/pkg/component/spec.go#L90-L95

Used here:
https://github.com/elastic/elastic-agent/blob/668fa22918fa330db979fc9fbd81ef3854deb4c2/pkg/component/runtime/command.go#L134-L142

I would suggest lowering this to 1s (or maybe even lower) for all of the Beats, otherwise there is a 10s period where there is no monitoring.

Reply:
In the original implementation it looks like the restart on output change was immediate: https://github.com/elastic/beats/pull/24907/files#diff-4ef060299d18c75dcda518842dfc08e553e892a4fc1e9b2ff4e4a03640be3fff

I would make this restart period as low as possible to minimize the time where the Beat isn't actually running when it should be running.